oda_api/api.py

```diff
@@ -181,23 +180,27 @@
             url = "https://www.astro.unige.ch/mmoda/dispatch-data"
 
         if host is not None:
-            logger.warning(
-                "please use 'url' instead of 'host' while providing dispatcher URL")
-            logger.warning(
-                "for now, we will adopt host, but in the near future it will not be done")
+            msg = '\n'
+            msg += '----------------------------------------------------------------------------\n'
+            msg += 'support for the parameter host will end soon \n'
+            msg += 'please use "url" instead of "host" while providing dispatcher URL \n'
+            msg += '----------------------------------------------------------------------------\n'
+            warnings.warn(msg, DeprecationWarning)
             self.url = host
 
             #TODO: disregard this, but leave parameter for compatibility
             if host.startswith('http'):
                 self.url = host
             else:
-                if protocol == 'http':
-                    self.url = "http://" + host
-                elif protocol == 'https':
-                    self.url = "https://" + host
-                else:
+                if protocol != 'http' and protocol != 'https':
                     raise UserError('protocol must be either http or https')
+                else:
+                    self.url = protocol + "://" + host
         else:
+            if not oda_api.misc_helpers.validate_url(url):
+                raise UserError(f'{url} is not a valid url, '
+                                f'please check it and try to issue again the request')
+
```

Comment:
people do not know what schema is, let's also write "...schema, i.e. url should start with http:// or https:// "